Makefile

```diff
@@ -71,21 +71,21 @@ build-aws: generate-functions
-	@DOCKER_BUILDKIT=1 docker build --target az --rm -t $(REMOTE_IMAGE_NAME):latest-az -t $(REMOTE_IMAGE_NAME):$(VERSION)-az -t $(REMOTE_IMAGE_NAME):$(COMMIT_HASH)-az --build-arg "FROM_IMAGE=$(FROM_IMAGE)" --build-arg "FROM_IMAGE_TAG=$(FROM_IMAGE_TAG)" --build-arg "VERSION=$(VERSION)-az" --build-arg "COMMIT_HASH=$(COMMIT_HASH)" .
-	@DOCKER_BUILDKIT=1 docker build --target final --rm -t $(REMOTE_IMAGE_NAME):latest -t $(REMOTE_IMAGE_NAME):$(VERSION) -t $(REMOTE_IMAGE_NAME):$(COMMIT_HASH) --build-arg "FROM_IMAGE=$(FROM_IMAGE)" --build-arg "FROM_IMAGE_TAG=$(FROM_IMAGE_TAG)" --build-arg "VERSION=$(VERSION)" --build-arg "COMMIT_HASH=$(COMMIT_HASH)" .
+	@DOCKER_BUILDKIT=1 docker build --target base --rm -t $(REMOTE_IMAGE_NAME):latest-minimal -t $(REMOTE_IMAGE_NAME):$(VERSION)-minimal -t $(REMOTE_IMAGE_NAME):$(COMMIT_HASH)-minimal --build-arg "FROM_IMAGE=$(FROM_IMAGE)" --build-arg "FROM_IMAGE_TAG=$(FROM_IMAGE_TAG)" --build-arg "VERSION=$(VERSION)-minimal" --build-arg "COMMIT_HASH=$(COMMIT_HASH)" --build-arg BUILDKIT_INLINE_CACHE=1 .
+	@DOCKER_BUILDKIT=1 docker build --target aws --rm -t $(REMOTE_IMAGE_NAME):latest-aws -t $(REMOTE_IMAGE_NAME):$(VERSION)-aws -t $(REMOTE_IMAGE_NAME):$(COMMIT_HASH)-aws --build-arg "FROM_IMAGE=$(FROM_IMAGE)" --build-arg "FROM_IMAGE_TAG=$(FROM_IMAGE_TAG)" --build-arg "VERSION=$(VERSION)-aws" --build-arg "COMMIT_HASH=$(COMMIT_HASH)" --build-arg BUILDKIT_INLINE_CACHE=1 .
+	@DOCKER_BUILDKIT=1 docker build --target az --rm -t $(REMOTE_IMAGE_NAME):latest-az -t $(REMOTE_IMAGE_NAME):$(VERSION)-az -t $(REMOTE_IMAGE_NAME):$(COMMIT_HASH)-az --build-arg "FROM_IMAGE=$(FROM_IMAGE)" --build-arg "FROM_IMAGE_TAG=$(FROM_IMAGE_TAG)" --build-arg "VERSION=$(VERSION)-az" --build-arg "COMMIT_HASH=$(COMMIT_HASH)" --build-arg BUILDKIT_INLINE_CACHE=1 .
+	@DOCKER_BUILDKIT=1 docker build --target final --rm -t $(REMOTE_IMAGE_NAME):latest -t $(REMOTE_IMAGE_NAME):$(VERSION) -t $(REMOTE_IMAGE_NAME):$(COMMIT_HASH) --build-arg "FROM_IMAGE=$(FROM_IMAGE)" --build-arg "FROM_IMAGE_TAG=$(FROM_IMAGE_TAG)" --build-arg "VERSION=$(VERSION)" --build-arg "COMMIT_HASH=$(COMMIT_HASH)" --build-arg BUILDKIT_INLINE_CACHE=1 .
 	@echo "Done!"
 
 .PHONY: push
 push:
-	@docker push --all-tags seiso/easy_infra
+	@docker push --all-tags $(REMOTE_IMAGE_NAME)
 
 .PHONY: update-apt
 update-apt:
 	@echo "Updating the apt package versions..."
 	@for package in $(APT_PACKAGES); do \
-		version=$$(docker run --rm easy_infra:latest /bin/bash -c "apt-get update &>/dev/null && apt-cache policy $${package} | grep '^  Candidate:' | awk -F' ' '{print \$$NF}'"); \
+		version=$$(docker run --rm $(REMOTE_IMAGE_NAME):latest-az /bin/bash -c "apt-get update &>/dev/null && apt-cache policy $${package} | grep '^  Candidate:' | awk -F' ' '{print \$$NF}'"); \
```

Comment:
To test locally I had to change REMOTE_IMAGE_NAME to LOCAL_IMAGE_NAME

Reply:
Good point, this PR depends on some other work I never pushed which needs to be done first.